odoo/addons/base/tests/test_ir_mail_server.py

```diff
@@ -136,6 +138,17 @@ def test_content_mail_body(self):
                     body_alternative = body_alternative.strip('\n')
             self.assertEqual(body_alternative, expected)
 
+    @mute_logger('odoo.sql_db')
+    def test_mail_server_auth_cert_requires_tls(self):
+        with self.assertRaises(psycopg2.errors.CheckViolation) as capture:
+            self.env['ir.mail_server'].create({
+                'name': 'test',
+                'smtp_host': 'smtp_host',
+                'smtp_encryption': 'none',
+                'smtp_authentication': 'certificate',
+            })
+        self.assertIn('ir_mail_server_certificate_requires_tls', capture.exception.args[0])
```

Comment:
Is it really necessary ? I mean it check how we create the final constraint name, but in the end maybe we just want to prevent the creation (don't care the actual constraint name) ?